src/data/polynomial/reverse.lean

```diff
@@ -0,0 +1,206 @@
+/-
+Copyright (c) 2020 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.polynomial.erase_lead
+
+/-!
+# Reverse of a univariate polynomial
+
+The main definition is `reverse`.  Applying `reverse` to a polynomial `f : polynomial R` produces
+the polynomial with a reversed list of coefficients, equivalent to `X^f.nat_degree * f(1/X)`.
+
+The main result is that `reverse (f * g) = reverse (f) * reverse (g)`, provided the leading
+coefficients of `f` and `g` do not multiply to zero.
+-/
+
+namespace polynomial
+
+open polynomial finsupp finset
+open_locale classical
+
+variables {R : Type*} [semiring R] {f : polynomial R}
+
+namespace rev
+/-- If `i ≤ N`, then `rev_at_fun N i` returns `N - i`, otherwise it returns `i`.
+
+This is the map used by the embedding `rev_at`.
+-/
+def rev_at_fun (N i : ℕ) : ℕ := ite (i ≤ N) (N-i) i
+
+@[simp] lemma rev_at_fun_invol {N i : ℕ} : rev_at_fun N (rev_at_fun N i) = i :=
```

Comment:
The linting error we get is:
> `#print polynomial.rev.rev_at_fun_invol` /- simp can prove this:
> `by simp only [polynomial.rev.rev_at_invol, polynomial.rev.rev_at_fun_eq]`
> One of the lemmas above could be a duplicate.
> If that's not the case try reordering lemmas or adding @[priority].
```

It says that this follows from `rev_at_fun_eq` and `rev_at_invol`, so we can drop the `@[simp]` tag:

```suggestion
lemma rev_at_fun_invol {N i : ℕ} : rev_at_fun N (rev_at_fun N i) = i :=
```